dspace-api/src/main/java/org/dspace/content/InProgressSubmission.java

```diff
@@ -18,13 +21,18 @@
  * @author Robert Tansley
  * @version $Revision$
  */
-public interface InProgressSubmission {
+public interface InProgressSubmission<ID extends Serializable> extends BrowsableDSpaceObject<ID> {
     /**
      * Get the internal ID of this submission
      *
      * @return the internal identifier
      */
-    Integer getID();
+    ID getID();
+
+    /**
+     * Update the submission, including the unarchived item.
+     */
+    void update() throws SQLException, AuthorizeException;
```

Comment:
I agree with @KevinVdV 's comment here. I'm not sure why we have an `update()` on a Database object.  It's especially odd since all the *implementations* of this `update()` that I see in this PR (e.g. in `WorkspaceItem` and `BasicWorkflowItem`) simply call back to the corresponding Service.  Couldn't we just drop this and just call the Service's `update()` directly?  It's unclear why we need this extra method.

Reply:
@abollini : Just drawing your attention to these comments ⬆️ , as I think it's another important bit of "cleanup".  It seems relatively minor to cleanup too (unless I'm missing something here).

Reply:
true! removed